sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -123,6 +126,24 @@ private[hive] object HiveShim {
     // for Serialization
     def this() = this(null)
 
+    override def hashCode(): Int = {
+        if (instance != null && instance.isInstanceOf[GenericUDFMacro]) {
```

Comment:
@yhuai Thanks !! Could users have a class that extends GenericUDFMacro ? Wouldn't we want to handle that case as well ?